components/content_settings/core/browser/brave_content_settings_pref_provider.cc

```diff
@@ -254,6 +303,43 @@ void BravePrefProvider::UpdateCookieRules(ContentSettingsType content_type,
     }
   }
 
+  if (prefs_->GetBoolean(kGoogleLoginControlType)) {
+    if (prefs_->GetInteger("profile.default_content_setting_values.cookies") == CONTENT_SETTING_BLOCK) {
+      AddCookieExceptions(CONTENT_SETTING_ALLOW, incognito);
+    }
+
+    if (prefs_->GetBoolean(prefs::kBlockThirdPartyCookies))  {
```

Comment:
the current cookie setting is not relevant, this is an override that is independent of cookie settings